src/coreComponents/physicsSolvers/inducedSeismicity/QuasiDynamicEQ.cpp

```diff
@@ -0,0 +1,282 @@
+    setInputFlag( InputFlags::OPTIONAL ).
+    setApplyDefaultValue( 1.0e-7 ).
+    setDescription( "Target slip incrmeent for timestep size selction" );
+}
+
+void QuasiDynamicEQ::postInputInitialization()
+{
+
+  // Initialize member stress solver as specified in XML input
+  if( !m_stressSolverName.empty() )
+  {
+    m_stressSolver = &this->getParent().getGroup< SolverBase >( m_stressSolverName );
+  }
+
+  SolverBase::postInputInitialization();
+}
+
+QuasiDynamicEQ::~QuasiDynamicEQ()
+{
+  // TODO Auto-generated destructor stub
+}
+
+void QuasiDynamicEQ::registerDataOnMesh( Group & meshBodies )
+{
+  SolverBase::registerDataOnMesh( meshBodies );
+
+  forDiscretizationOnMeshTargets( meshBodies, [&] ( string const &,
+                                                    MeshLevel & mesh,
+                                                    arrayView1d< string const > const & regionNames )
+  {
+    ElementRegionManager & elemManager = mesh.getElemManager();
+
+    elemManager.forElementSubRegions< SurfaceElementSubRegion >( regionNames,
+                                                                 [&]( localIndex const,
+                                                                      SurfaceElementSubRegion & subRegion )
+    {
+      subRegion.registerField< rateAndState::stateVariable >( getName() );
+      subRegion.registerField< rateAndState::slipRate >( getName() );
+      subRegion.registerField< rateAndState::stateVariable_n >( getName() );
+      subRegion.registerField< rateAndState::slipRate_n >( getName() );
+      subRegion.registerField< rateAndState::deltaSlip >( getName() );
+
+      if( !subRegion.hasWrapper( contact::slip::key() ))
+      {
+        string const labels[3] = { "normal", "tangent1", "tangent2" };
+
+        subRegion.registerField< contact::slip >( getName() );
+        subRegion.registerField< contact::traction >( getName() ).
+          setDimLabels( 1, labels ).
+          reference().resizeDimension< 1 >( 3 );
+
+        subRegion.registerWrapper< string >( viewKeyStruct::frictionLawNameString() ).
+          setPlotLevel( PlotLevel::NOPLOT ).
+          setRestartFlags( RestartFlags::NO_WRITE ).
+          setSizedFromParent( 0 );
+
+        string & frictionLawName = subRegion.getReference< string >( viewKeyStruct::frictionLawNameString() );
+        frictionLawName = SolverBase::getConstitutiveName< FrictionBase >( subRegion );
+        GEOS_ERROR_IF( frictionLawName.empty(), GEOS_FMT( "{}: FrictionBase model not found on subregion {}",
+                                                          getDataContext(), subRegion.getDataContext() ) );
+      }
+    } );
+  } );
+}
+
+real64 QuasiDynamicEQ::solverStep( real64 const & time_n,
+                                   real64 const & dt,
+                                   int const cycleNumber,
+                                   DomainPartition & domain )
+{
+  if( cycleNumber == 0 )
```

Comment:
todo?

Reply:
I am not sure how we will handle this until we change the location at which fractures are created.